File: 1. Beginner/Golang/bee-1118.go
package main

import "fmt"

func main () {
	for {
		grades := readInput()
		average := calculate(grades)
		printResult(average)
		action:= askForAction() 

		if !startOver(action) {
			return
		}
	}
}

func readInput () [2]float64 {
	counter, inp := 0, 0.0
	var grades [2]float64

	for counter < 2 {
		fmt.Scan(&inp)
		if ! isValid(inp) {
			fmt.Println("nota invalida")
			continue
		}
		grades[counter] = inp
		counter++
	}
	return grades
}

func isValid(grade float64) bool {
	return grade >= 0 && grade <= 10
}

func calculate(grades [2]float64) float64 {
	return (grades[0] + grades[1]) / 2
}

func printResult(average float64) {
	fmt.Printf("media = %.2f\n", average)
}

func askForAction() int {
	input := 0

	fmt.Println("novo calculo (1-sim 2-nao)")

	fmt.Scanf("%d", &input)

	return input
}

func startOver(action int) bool {
	if action == 1 {
		return true
	}

	if action == 2 {
		return false
	}

	action = askForAction()

	return startOver(action)
}

File: 1. Beginner/Java/bee-1118.java
import java.text.DecimalFormat;
import java.util.LinkedList;
import java.util.Scanner;

public class Main{
    public static void main(String[]args){
        Scanner sc = new Scanner(System.in);
        LinkedList<String> l =  new LinkedList<String>();
        DecimalFormat numberFormat = new DecimalFormat("#0.00"); 
        float grades[] = new float[2];
        int r = 0;
        
        do{
            int counter = 0, g = 0;
            for(; counter< grades.length;){
               float inp = sc.nextFloat();
               if (inp <= 10 && inp >= 0){
                   grades[counter] = inp;
                   counter++;
               }else g++;
            }
            for(int i = 0; i< g; i++){
                System.out.println("nota invalida");
            }
            System.out.println("media = " + numberFormat.format((grades[0]+grades[1])/2));
            for(;counter>=0; counter++){
                System.out.println("novo calculo (1-sim 2-nao)");
                r = sc.nextInt();
                if(r == 1 || r ==2) break;
            }
        }while(r == 1);
    }
}